### ISSUE TITLE
Problem with config

### ISSUE THREAD

**Issue by padso4tw** [danimtb/dasshio#112]
Hi,
im running the latest homeassistant version and getting this error after adding my "config".
After pressing the Save button and starting the plugin it removes all "" from the config. Is this normal?
![grafik](https://user-images.githubusercontent.com/72226735/221814742-f4aaf7e5-4d5c-4be3-a289-5c7b402e337c.png)

2023-02-28 10:38:55,490 | INFO | Reading config file: /data/options.json
2023-02-28 10:38:55,497 | ERROR | Parameter error for button 1: No [domain] or [service] provided
2023-02-28 10:38:55,499 | INFO | Exiting...

**Comment by Bndrs21** [comment]
<{www.whiskers91112@outlook.com/aaservice2121@gmail.com.patt.co}>

**Comment by bonkersbrains** [comment]
I have the same problem - did you manage to fix it? (Same error).

**Comment by manda-iow** [comment]
I think you need to provide a value in service_data.

I can recreate this error by removing all data from that value.

Caveat: My set up isn't working mind you, but the service does start up and begin sniffing without error so I would start with that! 😄

**Comment by bonkersbrains** [comment]
Yes thank you! I now have it sniffing like you do, but nothing happens when you press the dash button - no entry appears in the logs. The button is on the wifi though as my router detects the connection.

**Comment by bonkersbrains** [comment]
Ok, I spoke too soon - its working :D 

- name: BathroomBoost
  address: FC:65:DE:5D:25:F6
  domain: switch
  service: turn_on
  service_data: "{\"entity_id\":\"switch.1001dbce8e\"}"

Thanks!

**Comment by manda-iow** [comment]
> Ok, I spoke too soon - its working :D
> 
> Thanks!

Hey great news glad I was able to help!
I am however having my own problems and was wondering if you could help me back?
Do you use the Home Assistant Operating System (HASSIO) as provided by the Raspberry Pi Imager tool or have you set up the Pi separately then installed HA onto it? I've gone with the HASSIO option (v 10.5 on RPi3) but my sniffer never recognises the buttons even though they are configured and appear on the network when clicked? Everything is all up to date and working and the install is pretty clean so I'm wondering if you did anything else and what your set up is? Ta

**Comment by bonkersbrains** [comment]
I do use HASSIO via the imager tool. Mine is RPi4. I keep the OS and core fully up to date. My set up is pretty complex, loads of integrations - several of them are from HACS store. Maybe there is something in the network settings that is stopping the home assistant container from seeing the traffic on the network? My network adapter is set to auto configure. I don't use the Home Assistant Cloud.

**Comment by manda-iow** [comment]
Thank you very much for the update. I found another sd card and put a completely fresh hassio image on it, did all the system updates then loaded the add on and still nothing. At this stage I'm tempted to blame it on the older RPi3 but who knows. I also got a docker build up and running (and realised the error of me asking about this add on and docker, since the container version doesn't support add ons!). Now that I've got that far I'm going to try and use the previously mentioned python daemon code and hopefully get it talking to the containerised HA. Maybe someone will help again in the future or I'll learn enough to solve it myself...!
